addons/web_editor/static/src/components/media_dialog/icon_selector.js

```diff
@@ -0,0 +1,88 @@
+/** @odoo-module */
+
+import fonts from 'wysiwyg.fonts';
+import { SearchMedia } from './search_media';
+
+const { Component, useState, useEffect } = owl;
+
+export class IconSelector extends Component {
+    setup() {
+        fonts.computeFonts();
+
+        this.allFonts = fonts.fontIcons.map(({cssData, base}) => {
+            const uniqueIcons = Array.from(new Map(cssData.map(icon => {
+                const alias = icon.names.join(',');
+                const id = `${base}_${alias}`;
+                return [id, { ...icon, alias, id }];
+            })).values());
+            return { base, icons: uniqueIcons };
+        });
+
+        this.state = useState({
+            fonts: this.allFonts,
+            needle: '',
+        });
+
+        this.searchPlaceholder = this.env._t("Search a pictogram");
+
+        useEffect(() => {
+            const initWithMedia = async () => {
+                if (this.props.media) {
+                    const classes = this.props.media.className.split(/\s+/);
+                    const mediaFont = this.allFonts.filter(font => classes.includes(font.base))[0];
+                    if (mediaFont) {
+                        const selectedIcon = mediaFont.icons.filter(icon => {
+                            for (const name of icon.names) {
+                                if (classes.includes(name)) {
+                                    return true;
+                                }
+                            }
+                            return false;
+                        })[0];
+                        if (selectedIcon) {
+                            this.props.setInitialIconClasses(selectedIcon.names);
+                            await this.props.selectMedia(selectedIcon, { save: false });
+                        }
+                    }
+                }
+            };
+
+            initWithMedia();
+        }, () => []);
```

Comment:
(as a side note it's very weird that this logic is in this component in the first place)

Reply:
I agree... selectMedia and save are two different methods, I will change that and remove that save flag (and also, use the correct array methods)
For the test on the initial media given to the media dialog, I thought it could be the responsibility of the IconSelector component to do that, and I thought of using an effect because we don't want to execute that code when using the media dialog without the icons tab. But the effect should be defined with another dependency: `() => [this.props.media]` as it is acting on that. What do you think ?

Reply:
I already had the same concerns here: https://github.com/odoo/odoo/pull/89223/files/96254fa5d2a472cdf2357d30e228311a4d4f500b#r893478831 although I did not check as far as checking it should not even be done once mounted.

But yes, definitely something fishy here @younn-o